src/main/java/com/oracle/truffle/bf/BFV1.java

```diff
@@ -0,0 +1,157 @@
+
+package com.oracle.truffle.bf;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import com.oracle.truffle.api.RootCallTarget;
+import com.oracle.truffle.api.Truffle;
+import com.oracle.truffle.api.TruffleLanguage;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.nodes.ExplodeLoop;
+import com.oracle.truffle.api.nodes.Node;
+import com.oracle.truffle.api.nodes.RootNode;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Arrays;
+
+
+public class BFV1 extends BFImpl {
+
+    private BFParser.Operation[] operations;
+    private InputStream in;
+    private OutputStream out;
+    private RootCallTarget program;
+
+    static class Memory {
+        int[] cells;
+        int index;
+        final InputStream in;
+        final OutputStream out;
+
+        public Memory(InputStream in, OutputStream out) {
+            cells = new int[100];
+            this.in = in;
+            this.out = out;
+        }
+
+
+    }
+
+    @Override
+    public void prepare(BFParser.Operation[] operations, InputStream in, OutputStream out) {
+        this.operations = operations;
+        this.in = in;
+        this.out = out;
+        this.program = Truffle.getRuntime().createCallTarget(new BlockNode(TruffleLanguage.class, false, OpNode.convert(operations)));
+    }
+
+    @Override
+    public void run() throws IOException {
+        Memory memory = new Memory(in, out);
+        program.call(memory);
+    }
+
+    private static class BlockNode extends RootNode {
```

Comment:
BlockNode should probably not be a RootNode because you should not have multiple root nodes in one AST.